pyroSAR/drivers.py

```diff
@@ -1679,70 +1830,109 @@ def insert(self, scene_in, verbose=False, test=False):
+            session.commit()
         else:
             if verbose:
                 print('reverting temporary database changes...')
-            self.conn.rollback()
+            # roll back changes of the session
+            session.rollback()
         print('{} scenes registered regularly'.format(counter_regulars))
-        print('{} duplicates detected and registered'.format(counter_duplicates))
+        print('{} duplicates registered'.format(counter_duplicates))
+        if len(list_duplicates) != 0:
+            print('Scene(s) {} already registered as duplicate!'.format(list_duplicates))
     
     def is_registered(self, scene):
         """
         Simple check if a scene is already registered in the database.
 
         Parameters
         ----------
-        scene: str
+        scene: str or ID
+            the SAR scene
+
+        Returns
+        -------
+        bool
+            is the scene already registered?
+        """
+        id = scene if isinstance(scene, ID) else identify(scene)
+        # ORM query, where scene equals id.scene, return first
+        exists_data = self.Session().query(self.Data.scene).filter(self.Data.scene == id.scene).first()
+        exists_duplicates = self.Session().query(self.Duplicates.scene).filter(
+            self.Duplicates.scene == id.scene).first()
+        in_data = False
+        in_dup = False
+        if exists_data:
+            in_data = len(exists_data) != 0
+        if exists_duplicates:
+            in_dup = len(exists_duplicates) != 0
+        return in_data or in_dup
+    
+    def __is_registered_in_duplicates(self, scene):
+        """
+        Simple check if a scene is already registered in the database.
+
+        Parameters
+        ----------
+        scene: str or ID
             the SAR scene
 
         Returns
         -------
         bool
             is the scene already registered?
         """
-        return len(self.select(scene=scene)) != 0 or len(self.select_duplicates(scene=scene)) != 0
+        id = scene if isinstance(scene, ID) else identify(scene)
+        # ORM query as in is registered
+        exists_duplicates = self.Session().query(self.Duplicates.scene).filter(
+            self.Duplicates.scene == id.scene).first()
+        in_dup = False
+        if exists_duplicates:
+            in_dup = len(exists_duplicates) != 0
+        return in_dup
     
     def cleanup(self):
         """
         Remove all scenes from the database, which are no longer stored in their registered location
-        
+
         Returns
         -------
 
         """
-        cursor = self.conn.cursor()
         for table in ['data', 'duplicates']:
             missing = self.__select_missing(table)
             for scene in missing:
-                query = '''DELETE FROM {0} WHERE scene=?'''.format(table)
-                cursor.execute(query, (scene,))
-        self.conn.commit()
+                print(scene)
+                print(table)
```

Comment:
This could be a little confusing for the user if a lot of scenes are being removed. Please write something like
```Python
print('removing missing scene from database table {0}: {1}'.format(table, scene))
```